src/components/Calculator.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import NumberButtons from './NumberButtons';
+import OperatorButtons from './OperandButtons';
+import { expressionStorage } from '../store/store';
+import {
+  NUMBER_LIMIT,
+  ERROR_MSG,
+  CONFIRM_MSG,
+  OPERATOR,
+  OPERATOR_LIST,
+} from '../constants/constant';
+import AllClearButton from './AllCearButton';
+import Screen from './Screen';
+class Calculator extends Component {
+  constructor() {
+    super();
+    window.addEventListener('beforeunload', this.confirmExist);
+  }
+  state = {
+    sum: '',
+    prevNumber: [],
+    operator: '',
+    nextNumber: [],
+  };
+
+  componentDidMount() {
+    const expression = expressionStorage.getExpression();
+    if (!expression) {
+      return;
+    }
+    const { sum, prevNumber, operator, nextNumber } = expression;
+    this.setState({ sum, prevNumber, operator, nextNumber });
+  }
+
+  confirmExist = (e) => {
+    e.preventDefault();
+    e.returnValue = CONFIRM_MSG;
+    const { sum, prevNumber, operator, nextNumber } = this.state;
+    expressionStorage.setExpression({ sum, prevNumber, operator, nextNumber });
+  };
+
+  onClickNumber = (e) => {
+    const number = e.target.dataset.number;
+    const isPrev = this.state.operator === '';
+    try {
+      if (isPrev) {
+        if (this.state.prevNumber.length >= NUMBER_LIMIT) {
+          throw new Error(ERROR_MSG.OVER_NUMBER_LIMIT);
+        }
+        this.setState({ prevNumber: [...this.state.prevNumber, number] });
+        return;
+      }
+      if (this.state.nextNumber.length >= NUMBER_LIMIT) {
+        throw new Error(ERROR_MSG.OVER_NUMBER_LIMIT);
+      }
+      this.setState({ nextNumber: [...this.state.nextNumber, number] });
+    } catch ({ message }) {
+      alert(message);
+    }
```

Comment:
불필요한 try-catch 와 같이 댑스가 깊어지는 일은 피해주세요~!

```suggestion
    const { prevNumber, nextNumber } = this.state;
    
    if (prevNumber.length >= NUMBER_LIMIT || nextNumber.length >= NUMBER_LIMIT) {
      alert(ERROR_MSG.OVER_NUMBER_LIMIT);
      return;
    }
    
    if (isPrev) {
      this.setState({ prevNumber: [...this.state.prevNumber, number] });
      return;
    }
    
    this.setState({ prevNumber: [...this.state.prevNumber, number] });
```